allennlp/commands/train.py

```diff
@@ -502,7 +502,8 @@ def __init__(
         self.batch_weight_key = batch_weight_key
 
     def run(self) -> Dict[str, Any]:
-        return self.trainer.train()
+        with self.trainer as trainer:
+            return trainer.train()
```

Comment:
```suggestion
        with self.trainer:
            return self.trainer.train()
```
I don't really care one way or another, but I wanted to suggest this since it avoids an extra variable.